airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/DefaultAirbyteSource.java

```diff
@@ -95,7 +95,7 @@ public boolean isFinished() {
       return false;
     }
 
-    return !sourceProcess.isAlive() && !messageIterator.hasNext();
+    return !sourceProcess.isAlive();
```

Comment:
nit: Maybe this is slightly better?

```
  @Override
  public boolean isFinished() {
    Preconditions.checkState(sourceProcess != null);

    // As this check is done on every message read, it is important for this operation to be efficient.
    // Short circuit early to avoid checking the underlying process.
    // note: hasNext is blocking.
    return !messageIterator.hasNext() && !sourceProcess.isAlive();
  }
```